src/upgrades/mod.rs

```diff
@@ -261,4 +364,55 @@ mod tests {
 
         let _ = fs::remove_dir_all(work_dir);
     }
+
+    #[cfg(all(feature = "hsm", not(any(feature = "hsm-tests-kmip", feature = "hsm-tests-pkcs11"))))]
```

Comment:
Why is this excluded when either of those hsm-test features is enabled?

I seems to me that there would be no conflict in running this test again - and it's a bit friendlier on my IDE with all-features enabled :)

Reply:
It was excluded because it conceptually makes no sense (to me at least) to run this test when Krill has either of those special Rust features enabled as they assume a fresh installation that uses solely KMIP or PKCS#11, no OpenSSL at all and no migration from any prior set of OpenSSL keys as there will never be any prior state.